sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3192,6 +3193,69 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
       checkAnswer(df3, Array(Row(new java.math.BigDecimal("0.100000000000000000000000100"))))
     }
   }
+
+  test("SPARK-29037: For static partition overwrite, spark may give duplicate result.") {
```

Comment:
Once we are happy with the current approach, we may also port my concurrent write test case from https://github.com/apache/spark/pull/25739

Reply:
ok